src/Data/Provisioners/DBObjectProvisioner.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types=1);
+/**
+ * PHP Version 7
+ *
+ * @category   Data
+ * @package    Main
+ * @subpackage Data
+ * @author     Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\Data\Provisioners;
+
+/**
+ * A DBObejctProvisioner is an instance of ProvisionerInstance which
+ * takes an SQL query and the bind parameters to go with it, and
+ * executes it against the LORIS database.
+ *
+ * It also sets the fetch mode to PDO:FETCH_CLASS makes the statement
+ * return an instance of a given class name for each row.
+ *
+ * @category   Data
+ * @package    Main
+ * @subpackage Data
+ * @author     Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+abstract class DBObjectProvisioner extends \LORIS\Data\ProvisionerInstance
+{
+    private $_query;
```

Comment:
make checkstatic is returning this for me
```
 32 | WARNING | Property name "$_query" should not be prefixed with
    |         | an underscore to indicate visibility
 33 | WARNING | Property name "$_params" should not be prefixed with
    |         | an underscore to indicate visibility
 34 | WARNING | Property name "$_classname" should not be prefixed
    |         | with an underscore to indicate visibility

```

Reply:
@maltheism The `minor` and `major` branch currently have different linting rules because they're out of sync.

On `major`, that PHPCS error will be thrown because it is testing the `src/` directory with different coding standards as of #4335.

Unfortunately it seems that `major` did not get merged into `21.0-release` after #4335 was merged and so we are faced with this problem. 

At this point, these errors should be corrected on your new PR(s) on based on `major` if you're using these classes in them.

Reply:
This PR is sent to the minor branch. Travis is passing. @maltheism please rereview.